birdhouse/pavics-compose.sh

```diff
@@ -157,43 +95,36 @@ done
 export AUTODEPLOY_EXTRA_REPOS_AS_DOCKER_VOLUMES
 
 # we apply all the templates
-find ./config ./components ./optional-components $EXTRA_CONF_DIRS -name '*.template' |
+find $ALL_CONF_DIRS -name '*.template' |
   while read FILE
   do
     DEST=${FILE%.template}
     cat ${FILE} | envsubst "$VARS" | envsubst "$OPTIONAL_VARS" > ${DEST}
   done
 
 if [ x"$1" = x"up" ]; then
-  # this is external in docker-compose.yml so have to create here
-  # no error if already exist, just an error message
-  docker network create jupyterhub_network
-
-  # no error if already exist
-  # create externally so nothing will delete these data volume automatically
-  docker volume create jupyterhub_data_persistence  # jupyterhub db and cookie secret
-  docker volume create thredds_persistence  # logs, cache
-
-  if [ ! -f "$GEOSERVER_DATA_DIR/global.xml" ]; then
-    echo "fix GeoServer data dir permission on first run only, when data dir do not exist yet."
-    FIRST_RUN_ONLY=1 $COMPOSE_DIR/deployment/fix-geoserver-data-dir-perm
-  fi
-
-  for adir in ${EXTRA_CONF_DIRS}; do
+  for adir in $ALL_CONF_DIRS; do
     COMPONENT_PRE_COMPOSE_UP="$adir/pre-docker-compose-up"
     if [ -x "$COMPONENT_PRE_COMPOSE_UP" ]; then
       echo "executing '$COMPONENT_PRE_COMPOSE_UP'"
       sh -x "$COMPONENT_PRE_COMPOSE_UP"
     fi
   done
-
 fi
 
 COMPOSE_CONF_LIST="-f docker-compose.yml"
-for adir in ${EXTRA_CONF_DIRS}; do
-  if [ -f "$adir/docker-compose-extra.yml" ]; then
-    COMPOSE_CONF_LIST="${COMPOSE_CONF_LIST} -f $adir/docker-compose-extra.yml"
+for adir in $ALL_CONF_DIRS; do
+  if [ -f "$adir/docker-compose.yml" ]; then
+    COMPOSE_CONF_LIST="${COMPOSE_CONF_LIST} -f $adir/docker-compose.yml"
   fi
+  for extra_compose in "$adir"/docker-compose-extra-*.yml; do
+    service_name=$(basename "$extra_compose")
+    service_name=${service_name#docker-compose-extra-}
+    service_name=${service_name#.yml}
+    if echo "$CONFIGURED_COMPONENTS" | grep " $service_name "; then
```

Comment:
Oh and this `CONFIGURED_COMPONENTS` var is currently not set anywhere in the code?  If user will have to set it in `env.local`, add it to `env.local.example` so user know it is available.

Reply:
oops good catch with the `CONFIGURED_COMPONENTS` I think some extra code got lost in a merge, or maybe that was from an earlier draft. I'll double check and update that today.

Yes, I kept it so that additional components can be added as base paths.

Reply:
Now fixed